R/mock-data.R

```diff
@@ -0,0 +1,134 @@
+#' Generate mock gff 
+#'
+#' Quick function to generate a mock gff. 
+#' 
+#' @param n_genes Number of genes in mock gff file
+#' @param chromosome Chromosome value. Default=1.
+#' 
+#' @return data.frame containing mock gff data from chromosome 1.
+#' 
+#' @export
+#' 
+mock_gff <- function(n_genes=1000, chromosome=1){
+    
+    mock_gff <- data.frame(list(V1 = chromosome,
+                                V2 = "source",
+                                V3 = "gene",
+                                V4 = sort(sample(1e4:2e8, n_genes))))
+    mock_gff$V5 <- mock_gff$V4 + floor(rnorm(n_genes, 1500, 1000))
+    mock_gff[, c("V6", "V7", "V8", "V9")] <- "."
+    
+    return(mock_gff)
+}
+
+
+#' Generate mock vcf 
+#'
+#' Quick function to generate mock vcf file. Note this data has unrealistic
+#' population structure.
+#' 
+#' @param n_snps Number of SNPs in mock vcf file. Default=10,000.
+#' @param n_samples Number of samples in mock bulk data. Default=10.
+#' @param chromosome Chromosome value. Default=1.
+#' 
+#' @return data.frame containing mock gff data from chromosome 1.
+#'
+#' @export
+#' 
+mock_vcf <- function(n_snps=1e4, n_samples=10, chromosome=1){
+    
+    if (requireNamespace("vcfR", quietly = TRUE))
+    data(vcfR_example)
+    mock_vcf <- vcf
+    rm(gff, vcf, dna, pos=1)
+    mock_vcf@meta[2] <- "##source=\"Mock\""    
```

Comment:
There are not functions besides @ for extracting components from vcfR. There are a few bioconductor packages that have vcf objects (VariantAnnotation, TVTB, and VCFArray), however none of them have a "mock" function and none come with an example vcf object. So the only way to create a vcf object with any of those three software is to provide a vcf file yourself. I went with vcfR specifically because it already had the example data.  Happy to chat more about this. The data loaded in vcfR_example is just 3.4 MB.

Reply:
When I asked some Bioconductor people the general consensus was to use **VariantAnnotation**. I think it includes an example dataset. Once it is installed you should be able to do something like:

```r
 fl <- system.file("extdata", "chr22.vcf.gz", package="VariantAnnotation")
vcf <- readVcf(fl, "hg19")
```

Do you think it would be possible to use that instead? It might make things easier for people if we use a more standard format.

Alternatively how much of the VCF do you actually need? I'm wondering if it might be easier to just store a `data.frame` and extract the information for a VCF during estimation. I realise that's probably closer to what you had originally. I don't trust users to do the VCF to `data.frame` conversion themselves though so we will still have to deal with VCFs so maybe that doesn't make things any easier...

Sorry to make you look at this again but I think things will break if we need to change it later so I'd like to get it right now before anyone is using it.

Reply:
vcfR is replaced with VariantAnnotation